### PR TITLE
Simplify TestAnnotationHandler

### PR DESCRIPTION
**Pull request by dabrahams**
It doesn't seem like it's even an abstraction, so it could be removed entirely, but that's a bit
more of a project than I want to take on at this point.

### CODE REVIEW

**Comment by kyouko-taiga** [comment]
What's not an abstraction? The test annotation handler?

File: Tests/ValTests/Utils/CXXAnnotationHandler.swift [hunk 1]
```diff
@@ -4,38 +4,27 @@ import XCTest
...
-
   /// The C++ header.
   private let cxxHeader: String
 
   /// The C++ source.
   private let cxxSource: String
 
   /// The recorded issues.
-  private var issues: [XCTIssue] = []
+  private var issues_: [XCTIssue] = []
```
Comment (kyouko-taiga) [inline Tests/ValTests/Utils/CXXAnnotationHandler.swift:14]: IIRC, swift-format doesn't like names ending with an underscore.

File: Tests/ValTests/Utils/ValTestRunner.swift [hunk 2]
```diff
@@ -15,22 +15,17 @@ extension ValTestRunner {
   /// Runs the test cases found in `Self.testCaseDirectoryPath` with `runSteps` and handle the
   /// result of each case with an instance of `handler`.
   func runValTests<Handler: TestAnnotationHandler>(
-    handlingResultsWith hander: Handler.Type,
-    _ runSteps: (_ name: String, _ source: SourceFile) -> Handler.Configuration
+    _ runSteps: (_ xcTestName: String, _ source: SourceFile) throws -> Handler
```
Comment (kyouko-taiga) [inline Tests/ValTests/Utils/ValTestRunner.swift:18]: Nice change!

**Review by kyouko-taiga — APPROVED** [review]
Approved, but I think there are some swift-format issues.

File: Tests/ValTests/Utils/CXXAnnotationHandler.swift (hunk 1 above)
Comment (dabrahams) [reply Tests/ValTests/Utils/CXXAnnotationHandler.swift:14]: That's very silly; they catch it as not obeying lowerCamelCase, as though it's somehow snake_case to end with an underscore.  Fortunately I can start with one.

**Comment by dabrahams** [comment]
Yes, when all the complexity is unwound `TestAnnotationHandler` seems like it's ultimately just a function taking test annotations and returning XCTest issues.  Maybe that's an abstraction; I'm not entirely sure.

**Pull request merged**